File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(text) {
    this.matrix = []
    this.inverseMatrix = []
    const lines = text.split('\n');
    lines.forEach((line, lineIndex) => {
      this.matrix[lineIndex] = line.split(' ');
    });
    console.log(this.matrix)
  }

  get rows() {
    return this.matrix;
  }

  inverseMatrix = () => {
    this.inverseMatrix = this.matrix
    for (let i = 0; i < this.matrix.length; i++) {
      for (let j = 0; j < this.matrix[i].length; j++) {
        this.inverseMatrix[j][i] = this.matrix[i][j];
      }
    }
  }

  get columns() {
    const cols = [];
    const rows = this.rows();
    rows.forEach((row, i) => {
      row[i].forEach((el, j) => {
        
      });
    });

    return cols;
  }
}

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string : str):
        self.rows = []
        self.cols = []
        rows_string = matrix_string.split("\n")
        self.get_rows_from_string(rows_string)
        self.get_columns_from_rows()



    def row(self, index : int) -> []:
        return self.rows[index - 1]

    def column(self, index : int) -> []:
        return self.cols[index - 1]

    def get_rows_from_string(self, rows_string : str):
        for row_string in rows_string:
            row_string = row_string.strip()
            row_string = row_string.split(" ")
            self.rows.append([int(num) for num in row_string])

    def get_columns_from_rows(self):
        if len(self.rows) == 0:
            return
        for i in range(len(self.rows[0])):
            self.cols.append([row[i] for row in self.rows])
